src/script/lua_api/l_camera.cpp

```diff
@@ -61,12 +61,26 @@ int LuaCamera::l_set_camera_mode(lua_State *L)
 	if (!lua_isnumber(L, 2))
 		return 0;
 
-	camera->setCameraMode((CameraMode)((int)lua_tonumber(L, 2)));
+	CameraMode mode = (CameraMode)((int)lua_tonumber(L, 2));
+
+	if ((mode == CAMERA_MODE_CUSTOM) && getClient(L)->checkCSMRestrictionFlag(CSMRestrictionFlags::CSM_RF_CAMERA_CONTROL))
+		return 0;
+
+	camera->setCameraMode(mode);
 	playercao->setVisible(camera->getCameraMode() > CAMERA_MODE_FIRST);
 	playercao->setChildrenVisible(camera->getCameraMode() > CAMERA_MODE_FIRST);
 	return 0;
 }
 
+int LuaCamera::l_set_camera_custom_state(lua_State *L)
+{
+	Camera *camera = getobject(L, 1);
+	if (!camera)
+		return 0;
+	camera->setCameraCustomState(read_v3f(L, 2), read_v3f(L, 3), read_v3f(L, 4));
```

Comment:
Might it be a good idea to actually check if all the params have been provided here, and then call this function?

Reply:
I'd agree, but then they could be invalid anyway. Is there a `verify_v3f` function?

Reply:
AFAIK, there's no such function (yet?), but checking if the params even exist might be a good start, IMHO.

Reply:
So there is actually such a function; `check_v3f`. Will use that.

Reply:
This was resolved in 9e6e680.